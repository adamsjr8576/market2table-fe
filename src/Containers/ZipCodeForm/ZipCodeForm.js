import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import './ZipCodeForm.scss';
import { addZipCode, addMarkets } from '../../actions';
import { getMarketsByZip } from '../../apiCalls';

export const ZipCodeForm = ({ path }) => {

  const [zipCodeInput, setzipCodeInput] = useState('');
  const [hasError, setHasError] = useState(false);
  const dispatch = useDispatch();
  const zipCode = useSelector(state => state.zipCode);

  const handleSubmit = () => {
    if (zipCodeInput.length === 5) {
      getMarketsByZip(zipCodeInput)
        .then(markets => {
          dispatch(addMarkets(markets));
        })
      .catch(error => console.log(error))
      dispatch(addZipCode(zipCodeInput));
      setzipCodeInput('');
    } else {
      setHasError(true);
      setzipCodeInput('');
    }
  }

  let opacity = 0;
    if (path.includes('markets')) {
      opacity = .9;
    }

  const handleChange = (value) => {
    if (value.length > 5) {
      value = value.slice(0, 5);
    }
    setzipCodeInput(value);
    setHasError(false);
  }

  const zipInputHandler = event => {
    if (event.key === "Enter") {
      event.preventDefault()
      handleSubmit()
      }
    }

  if (zipCode > 0) {
    return (
      <Redirect to='/markets' />
    )
  }
  return (
    <section className='zip-code-form-container' style={{backgroundImage: `linear-gradient(rgba(255, 255, 255, ${opacity}), rgba(255, 255, 255, ${opacity}))` }}>
      <form className='zip-code-form'>
        <label className='zip-code-input-label' htmlFor='zipCode'>
          Enter a zip code to find farmer's markets near you:
        </label>
        <input onChange={(e) => handleChange(e.target.value)} onKeyDown={event => zipInputHandler(event)} value={zipCodeInput} className="zip-code-input" type="number" placeholder="insert zip code" id="zipCode" name="zipCode" />
        <div className='error-message-container'>
          <p hidden={!hasError} className='error-message'>Error: Please enter a 5 digit zip code</p>
        </div>
        <button type='button' onClick={handleSubmit} className='submit-zip-code-button'>Find!</button>
      </form>
    </section>
  )
}

export default ZipCodeForm;
