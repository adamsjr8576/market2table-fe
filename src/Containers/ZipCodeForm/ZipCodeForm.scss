@import './src/variables';

.zip-code-form-container {
  width: 100%;
  height: 85vh;
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
}

.zip-code-form {
  color: white;
  height: 250px;
  width: 45%;
  min-width: 400px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  background-color: rgba(0, 0, 0, .6);
  border-radius: 20px;
  border: 3px solid khaki;
}

.zip-code-input-label {
  font-size: 1.5em;
  letter-spacing: 1px;
  margin: 0 25px;
  color: cornsilk;
}

.zip-code-input {
  font-size: 1.5em;
  // border-radius: 20px;
  outline: 0;
  border: none;
  letter-spacing: 1px;
  padding: 7px;
  text-align: center;
  appearance: none;
  background-color: transparent;
  border-bottom: 2px solid khaki;
  caret-color: khaki;
  color: cornsilk;
  letter-spacing: 1px;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.error-message-container {
  height: 10px;
}

.error-message {
  color: red;
  margin: 0;
}

.submit-zip-code-button {
  font-size: 1.5em;
  border-radius: 20px;
  letter-spacing: 1px;
  padding: 5px;
  width: 20%;
  min-width: 100px;
  cursor: pointer;
  outline: 0;
  color: cornsilk;
  background-color: transparent;
  border: 2px solid khaki;
}

.submit-zip-code-button:hover,
.submit-zip-code-button:focus {
  transform: scale(1.05);
}
