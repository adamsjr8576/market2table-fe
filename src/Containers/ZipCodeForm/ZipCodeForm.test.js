import React from 'react';
import { shallow } from 'enzyme';
import ZipCodeForm from './ZipCodeForm';
import { useSelector, useDispatch } from 'react-redux';

const mockState = {
    zipCode: '80211',
  }
jest.mock("react-redux", () => ({
    useSelector: () => mockState,
    useDispatch: () => jest.fn()
}));

describe('ZipCodeForm', () => {
    let mockAddZipCode = jest.fn()
    let mockAddMarkets = jest.fn()
    it('should match the ZipCodeForm snapshot', () => {
        let wrapper = shallow(<ZipCodeForm
            path='/markets'
            opacity={0}
            zipCode={mockState.zipCode}
            addMarkets={mockAddMarkets}
            addZipCode={mockAddZipCode}
          />);
        expect(wrapper).toMatchSnapshot();
    })

    it('should match the ZipCodeForm snapshot', () => {
        let wrapper = shallow(<ZipCodeForm
            path='/foster'
            opacity={.9}
            zipCode={mockState.zipCode}
            addMarkets={mockAddMarkets}
            addZipCode={mockAddZipCode}
          />);
        expect(wrapper).toMatchSnapshot();
    })
});
