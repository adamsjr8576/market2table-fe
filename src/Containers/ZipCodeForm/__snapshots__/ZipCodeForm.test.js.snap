// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZipCodeForm should match the ZipCodeForm snapshot 1`] = `
<section
  className="zip-code-form-container"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(255, 255, 255, 0.9), rgba(255, 255, 255, 0.9))",
    }
  }
>
  <form
    className="zip-code-form"
  >
    <label
      className="zip-code-input-label"
      htmlFor="zipCode"
    >
      Enter a zip code to find farmer's markets near you:
    </label>
    <input
      className="zip-code-input"
      id="zipCode"
      name="zipCode"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="zip code..."
      type="number"
      value=""
    />
    <div
      className="error-message-container"
    >
      <p
        className="error-message"
        hidden={true}
      >
        Error: Please enter a 5 digit zip code
      </p>
    </div>
    <button
      className="submit-zip-code-button"
      onClick={[Function]}
      type="button"
    >
      Find!
    </button>
  </form>
</section>
`;

exports[`ZipCodeForm should match the ZipCodeForm snapshot 2`] = `
<section
  className="zip-code-form-container"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(255, 255, 255, 0), rgba(255, 255, 255, 0))",
    }
  }
>
  <form
    className="zip-code-form"
  >
    <label
      className="zip-code-input-label"
      htmlFor="zipCode"
    >
      Enter a zip code to find farmer's markets near you:
    </label>
    <input
      className="zip-code-input"
      id="zipCode"
      name="zipCode"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="zip code..."
      type="number"
      value=""
    />
    <div
      className="error-message-container"
    >
      <p
        className="error-message"
        hidden={true}
      >
        Error: Please enter a 5 digit zip code
      </p>
    </div>
    <button
      className="submit-zip-code-button"
      onClick={[Function]}
      type="button"
    >
      Find!
    </button>
  </form>
</section>
`;
